src/treq/response.py

```diff
@@ -95,3 +97,25 @@ def cookies(self):
                 jar.set_cookie(cookie)
 
         return jar
+
+    def fail_for_status(self):
```

Comment:
The name `fail_for_status` was chosen for consistency with the name `raise_for_status` in `requests`. Since '[mimicking the requests API is not a goal](https://treq.readthedocs.io/en/latest/)' I will use `check_status`.